.github/workflows/CI.yml

```diff
@@ -23,10 +23,14 @@ jobs:
         cmake_build_type: ['Debug', 'Release']
     runs-on: ubuntu-20.04
     container: ghcr.io/ornl-mdf/ci-containers/ubuntu:latest
+    continue-on-error: true
     steps:
       - name: Checkout 3DThesis
+        id: checkout
         uses: actions/checkout@v3
       - name: Build 3DThesis
+        if: steps.checkout.outcome == 'success'
```

Comment:
Shouldn't need to do this - it will stop if a step fails

Reply:
> This is intended to work together with the `if: steps.build.outcome == 'success'` parts on the individual steps. From my understanding, if the code gets checked out and built, then following steps can run and fail/pass independent of each other. If any step fails, the job still fails (when `continue-on-error` is given at the job-level).

I think it is necessary given my attempt at this. There may be a cleaner way to do this that I am not aware of.

Reply:
`continue-on-error` allows other jobs to continue (generally not helpful but sometimes), but the current job can't if a previous step failed

Reply:
Ahh, okay. I just removed this, so it will just stop if it runs into an error at any point. Someone can use the bash script to debug locally if needed.